web/src/main/java/org/cbioportal/web/util/appliers/StructuralVariantSubFilterApplier.java

```diff
@@ -0,0 +1,112 @@
+package org.cbioportal.web.util.appliers;
+
+import org.cbioportal.model.MolecularProfile;
+import org.cbioportal.model.StructVarFilterQuery;
+import org.cbioportal.model.StudyViewStructuralVariantFilter;
+import org.cbioportal.service.MolecularProfileService;
+import org.cbioportal.service.StructuralVariantService;
+import org.cbioportal.web.parameter.SampleIdentifier;
+import org.cbioportal.web.parameter.StudyViewFilter;
+import org.cbioportal.web.util.StudyViewFilterUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Component
+public class StructuralVariantSubFilterApplier implements StudyViewSubFilterApplier {
+
+    @Autowired
+    private MolecularProfileService molecularProfileService;
+
+    @Autowired
+    private StructuralVariantService structuralVariantService;
+
+    @Autowired
+    private StudyViewFilterUtil studyViewFilterUtil;
+    
+    @Override
+    public List<SampleIdentifier> filter(List<SampleIdentifier> toFilter, StudyViewFilter filters) {
```

Comment:
This is probably part of a larger discussion, but we will eventually need to translate this filter to SQL if we want to migrate study view filter related logic into Clickhouse, right?